coreos-ostree-importer/coreos_ostree_importer.py

```diff
@@ -0,0 +1,280 @@
+EXAMPLE_MESSAGE_BODY = json.loads(
+    """
+{
+    "build_id": "30.20190905.0",
+    "stream": "testing",
+    "basearch": "x86_64",
+    "commit": "s3://fcos-builds/prod/streams/testing/builds/30.20190905.0/x86_64/ostree-commit.tar",
+    "checksum": "sha256:d01db6939e7387afa2492ac8e2591c53697fc21cf16785585f7f1ac0de692863",
+    "ostree_ref": "fedora/x86_64/coreos/testing",
+    "ostree_checksum": "b4beca154dab3696fd04f32ddab818102caa9247ec3192403adb9aaecc991bd9",
+    "target_repo": "prod"
+}
+"""
+)
+
+KNOWN_OSTREE_REPOS = {
+    "prod": "/mnt/koji/ostree/repo",
+    "compose": "/mnt/koji/compose/ostree/repo",
+}
+
+# Given a repo (and thus an input JSON) analyze existing koji tag set
+# and tag in any missing packages
+class Consumer(object):
+    def __init__(self):
+        logger.info(
+            "Processing messages with topic: %s" % FEDORA_MESSAGING_TOPIC_LISTEN
+        )
+
+    def __call__(self, message: fedora_messaging.api.Message):
+        # Catch any exceptions and don't raise them further because
+        # it will cause /usr/bin/fedora-messaging to crash and we'll
+        # lose the traceback logs from the container
+        try:
+            self.process(message)
+            send_message(msg=message.body, status="SUCCESS")
+        except Exception as e:
+            logger.error("Caught Exception!")
+            logger.error("###################################")
+            traceback.print_exc()
+            logger.error("###################################")
+            logger.error("Replying with a FAILURE message...")
+            send_message(msg=message.body, status="FAILURE")
+            logger.error("\t continuing...")
+            pass
+
+    def process(self, message: fedora_messaging.api.Message):
+        logger.debug(message.topic)
+        logger.debug(message.body)
+
+        # Grab the raw message body and parse out pieces
+        msg = message.body
+        basearch = msg["basearch"]
+        build_id = msg["build_id"]
+        checksum = msg["checksum"]
+        commit_url = msg["commit"]
+        ostree_checksum = msg["ostree_checksum"]
+        ostree_ref = msg["ostree_ref"]
+        stream = msg["stream"]
+        target_repo = msg["target_repo"]
+
+        # Qualify arguments
+        if not checksum.startswith("sha256:"):
+            raise Exception("checksum value must start with sha256:")
+        if target_repo not in KNOWN_OSTREE_REPOS.keys():
+            raise Exception(f"Provided target repo is unknown: {target_repo}")
+
+        sha256sum = checksum[7:]
+        target_repo_path = KNOWN_OSTREE_REPOS[target_repo]
+        source_repo_path = None
+
+        # Parse the message
+
+        # Check the target repo to make sure it exists
+        # and is accessible
+        if not ostree_repo_exists(target_repo_path):
+            raise Exception(f"OSTree repo does not exist at {target_repo_path}")
```

Comment:
Can we just check this upfront at init time?

Reply:
will do

Reply:
will do